Sources/Swift/Protocol/Codable/SentryEventCodable.swift

```diff
@@ -0,0 +1,107 @@
+        self.startTimestamp = try self.decodeTimeIntervalSince1970(from: container, forKey: .startTimestamp)
+
+        if let rawLevel = try container.decodeIfPresent(String.self, forKey: .level) {
+            let level = SentryLevelHelper.levelForName(rawLevel)
+            SentryLevelBridge.setBreadcrumbLevelOn(self, level: level.rawValue)
+        } else {
+            SentryLevelBridge.setBreadcrumbLevelOn(self, level:
+            SentryLevel.none.rawValue)
+        }
+
+        self.platform = try container.decode(String.self, forKey: .platform)
+        self.logger = try container.decodeIfPresent(String.self, forKey: .logger)
+        self.serverName = try container.decodeIfPresent(String.self, forKey: .serverName)
+        self.releaseName = try container.decodeIfPresent(String.self, forKey: .releaseName)
+        self.dist = try container.decodeIfPresent(String.self, forKey: .dist)
+        self.environment = try container.decodeIfPresent(String.self, forKey: .environment)
+        self.transaction = try container.decodeIfPresent(String.self, forKey: .transaction)
+        self.type = try container.decodeIfPresent(String.self, forKey: .type)
+        self.tags = try container.decodeIfPresent([String: String].self, forKey: .tags)
+
+        self.extra = decodeArbitraryData {
+            try container.decodeIfPresent([String: ArbitraryData].self, forKey: .extra)
+        }
+        self.sdk = decodeArbitraryData {
+            try container.decodeIfPresent([String: ArbitraryData].self, forKey: .sdk)
+        }
+
+        self.modules = try container.decodeIfPresent([String: String].self, forKey: .modules)
+        self.fingerprint = try container.decodeIfPresent([String].self, forKey: .fingerprint)
+        self.user = try container.decodeIfPresent(User.self, forKey: .user)
+        
+        self.context = decodeArbitraryData {
+            try container.decodeIfPresent([String: [String: ArbitraryData]].self, forKey: .context)
+        }
+
+        if let rawThreads = try container.decodeIfPresent([String: [SentryThread]].self, forKey: .threads) {
+            self.threads = rawThreads["values"]
+        }
+            
+        if let rawExceptions = try container.decodeIfPresent([String: [Exception]].self, forKey: .exception) {
+            self.exceptions = rawExceptions["values"]
+        }
+        
+        self.stacktrace = try container.decodeIfPresent(SentryStacktrace.self, forKey: .stacktrace)
+        
+        if let rawDebugMeta = try container.decodeIfPresent([String: [DebugMeta]].self, forKey: .debugMeta) {
+            self.debugMeta = rawDebugMeta["images"]
+        }
+        
+        self.breadcrumbs = try container.decodeIfPresent([Breadcrumb].self, forKey: .breadcrumbs)
+        self.request = try container.decodeIfPresent(SentryRequest.self, forKey: .request)
+    }
+    
+    private func decodeTimeIntervalSince1970(
```

Comment:
`l`: Should this be a decoding helper outside of this type?

Reply:
We're currently only using it in this class. Once we use it somewhere else, I would move it. Is that OK for you?

Reply:
I'm going to follow up with another PR on this comment if required.

Reply:
If fixed this problem with https://github.com/getsentry/sentry-cocoa/pull/4802.